src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java

```diff
@@ -231,43 +236,59 @@ private void merge(ProcessConfig processConfig, ZonedDateTime dateTime, String d
         }
     }
 
-    private List<IgmEntity> findValidatedIgms(ZonedDateTime dateTime, UUID processUuid) {
-        // Use of UTC Zone to store in cassandra database
+    @Transactional(readOnly = true)
+    public List<IgmEntity> doFindValidatedIgms(UUID processUuid, LocalDateTime localDateTime) {
+        return igmRepository.findByProcessUuidAndDate(processUuid, localDateTime).stream().map(entity -> {
+            @SuppressWarnings("unused")
+            int ignoreSize = entity.getBoundaries().size();
+            return entity;
+        }).collect(Collectors.toList());
+    }
+
+    public List<IgmEntity> findValidatedIgms(ZonedDateTime dateTime, UUID processUuid) {
+        // Use of UTC Zone to store in database
         LocalDateTime localDateTime = LocalDateTime.ofInstant(dateTime.toInstant(), ZoneOffset.UTC);
+        return self.doFindValidatedIgms(processUuid, localDateTime).stream()
+            .filter(mergeEntity -> mergeEntity.getStatus().equals(IgmStatus.VALIDATION_SUCCEED.name()))
+            .collect(Collectors.toList());
+    }
+
+    @Transactional(readOnly = true)
+    public List<IgmEntity> doFindAllIgms() {
+        return igmRepository.findAll().stream().map(entity -> {
+            @SuppressWarnings("unused")
+            int ignoreSize = entity.getBoundaries().size();
+            return entity;
+        }).collect(Collectors.toList());
+    }
 
-        return igmRepository.findByProcessUuidAndDate(processUuid, localDateTime).stream()
-                .filter(mergeEntity -> mergeEntity.getStatus().equals(IgmStatus.VALIDATION_SUCCEED.name()))
-                .collect(Collectors.toList());
+    public List<IgmEntity> findAllIgms() {
+        return self.doFindAllIgms();
```

Comment:
we can avoid self invocation, by annotating findAllIgms as transactional

Reply:
Done